Userland/Libraries/LibC/wstdio.cpp

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2021, Ali Mohammad Pur <mpfard@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Assertions.h>
+#include <AK/StringBuilder.h>
+#include <AK/Types.h>
+#include <bits/stdio_file_implementation.h>
+#include <stdio.h>
+#include <wchar.h>
+
+static_assert(AssertSize<wchar_t, sizeof(u32)>());
+
+extern "C" {
+
+wint_t fgetwc(FILE* stream)
+{
+    VERIFY(stream);
+    Array<u8, 4> underlying;
+    auto underlying_bytes = underlying.span();
+    size_t encoded_length = 0;
+    size_t bytes_read = 0;
+    do {
+        size_t nread = fread(underlying_bytes.offset_pointer(bytes_read), 1, 1, stream);
+        if (nread != 1)
+            return WEOF;
+        ++bytes_read;
+        if (bytes_read == 1) {
+            if (underlying[0] >> 7 == 0)
+                encoded_length = 1;
+            else if (underlying[0] >> 5 == 6)
+                encoded_length = 2;
+            else if (underlying[0] >> 4 == 0xe)
+                encoded_length = 3;
+            else
+                encoded_length = 4;
+        }
```

Comment:
Won't things go wrong here if we're sent invalid UTF-8?

Reply:
I decided to ignore encoding errors as I didn't have much time to think about them yesterday, I'll take a look.

Reply:
Now it just returns `WEOF` and sets `errno` to `EILSEQ`.